pkg/kubectl/resource_printer_test.go

```diff
@@ -164,6 +164,23 @@ func TestPrintTemplate(t *testing.T) {
 	}
 }
 
+func TestPrintJSONPath(t *testing.T) {
+	buf := bytes.NewBuffer([]byte{})
+	printer, found, err := GetPrinter("jsonpath", "{.metadata.name}")
```

Comment:
I understand that this is patterned after TestPrintTemplate, but what I'd like to see is a table-driven test that exercises more cases than just a single field. 

An example of a table-driven test:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate_test.go#L52

The test could be expanded in a subsequent PR, though.